src/main/webapp/app/exercises/programming/manage/grading/tasks/programming-exercise-grading-tasks-table.component.ts

```diff
@@ -138,13 +138,13 @@ export class ProgrammingExerciseGradingTasksTableComponent implements OnInit {
     };
 
     private compareNumForAttribute = (attributeKey: string): TaskComparator => {
-        return (a, b) => {
+        return (a: ProgrammingExerciseTask, b: ProgrammingExerciseTask) => {
             return ((a[attributeKey] as number) ?? 0) - ((b[attributeKey] as number) ?? 0);
         };
     };
 
     private compareStringForAttribute = (attributeKey: string): TaskComparator => {
-        return (a, b) => {
+        return (a: ProgrammingExerciseTask, b: ProgrammingExerciseTask) => {
```

Comment:
Is that right? The corresponding type of `TaskComparator` is:

```ts
type TaskComparator = (a: ProgrammingExerciseTask | ProgrammingExerciseTestCase, b: ProgrammingExerciseTask | ProgrammingExerciseTestCase) => number;
```

Suggesting to me that we also compare test cases with this method

Reply:
@SimonEntholzer